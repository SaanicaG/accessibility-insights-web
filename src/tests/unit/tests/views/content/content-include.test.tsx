// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { render } from '@testing-library/react';
import * as React from 'react';

import {
    expectMockedComponentPropsToMatchSnapshots,
    mockReactComponents,
} from 'tests/unit/mock-helpers/mock-module-helpers';
import { Content } from 'views/content/content';
import { ContentInclude } from 'views/content/content-include';
import { ContentPage } from 'views/content/content-page';
import { ContentActionMessageCreator } from '../../../../../common/message-creators/content-action-message-creator';

jest.mock('views/content/content');
describe('ContentInclude', () => {
    mockReactComponents([Content]);
    const content = {
        for: {
            testing: ContentPage.create(() => 'CONTENT FOR TESTING' as any),
        },
    };

    const applicationTitle = 'THE_APPLICATION_TITLE';
    const deps = {
        textContent: {
            applicationTitle,
        },
        contentProvider: ContentPage.provider(content),
        contentActionMessageCreator: {} as any as ContentActionMessageCreator,
    };

    it('renders from content', () => {
        const renderResult = render(<ContentInclude deps={deps} content={content.for.testing} />);
        expect(renderResult.asFragment()).toMatchSnapshot();
        expectMockedComponentPropsToMatchSnapshots([Content]);
    });

    it('renders from path', () => {
        const renderResult = render(<ContentInclude deps={deps} content={'for/testing'} />);
        expect(renderResult.asFragment()).toMatchSnapshot();
        expectMockedComponentPropsToMatchSnapshots([Content]);
    });
});
