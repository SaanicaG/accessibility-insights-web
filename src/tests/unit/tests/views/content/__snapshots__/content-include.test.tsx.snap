// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentInclude renders from content 1`] = `
<DocumentFragment>
  <div
    class="content"
  >
    CONTENT FOR TESTING
  </div>
</DocumentFragment>
`;

exports[`ContentInclude renders from content: Content props 1`] = `undefined`;

exports[`ContentInclude renders from path 1`] = `
<DocumentFragment>
  <div
    class="content"
  >
    CONTENT FOR TESTING
  </div>
</DocumentFragment>
`;

exports[`ContentInclude renders from path: Content props 1`] = `undefined`;
